app/assets/v2/js/grants/detail.js

```diff
@@ -111,27 +110,42 @@ $(document).ready(function() {
             gas: 3000000,
             gasPrice: web3.utils.toHex($('#gasPrice').val() * Math.pow(10, 9))
           }).on('transactionHash', function(transactionHash) {
-            grant_cancel_tx_id = $('#grant_cancel_tx_id').val();
             const linkURL = etherscan_tx_url(transactionHash);
 
             document.issueURL = linkURL;
             $('#transaction_url').attr('href', linkURL);
             $('.modal .close').trigger('click');
             enableWaitState('#grants-details');
+
+            $.ajax({
+              type: 'post',
+              url: '',
+              data: {
+                'grant_cancel_tx_id': transactionHash,
+                'csrfmiddlewaretoken': $("#grants-details input[name='csrfmiddlewaretoken']").val()
+              },
+              success: function(json) {
+                console.log('Your grant_cancel call successfully saved');
+              },
+              error: function() {
+                _alert({ message: gettext('Canceling your grant failed to save. Please try again.') }, 'error');
+              }
+            });
           })
           .on('confirmation', function(confirmationNumber, receipt) {
             $.ajax({
               type: 'post',
               url: '',
               data: {
-                'contract_address': contract_address,
-                'grant_cancel_tx_id': grant_cancel_tx_id
+                'csrfmiddlewaretoken': $("#grants-details input[name='csrfmiddlewaretoken']").val(),
+                'confirmed': true
               },
               success: function(json) {
+                console.log('Your cancel_grant tx successfully confirmed on chain');
                 window.location.reload(false);
               },
               error: function() {
-                _alert({ message: gettext('Canceling your grant failed to save. Please try again.') }, 'error');
+                _alert({ message: gettext('Your cancel_grant tx failed. Please try again.') }, 'error');
```

Comment:
Should we rename this to! `Canceling grant failed. Please try again`  cause `cancel_grant` seems a lil cryptic for an alert to the enduser

Reply:
Sure. I will change this.